sbin/build.sh

```diff
@@ -459,6 +463,42 @@ executeTemplatedFile() {
     exit 2
   fi
 
+  # Restore exit behavior
+  set -eu
+}
+
+createOpenJDKFailureLogsArchive() {
+    echo "OpenJDK make failed, archiving make failed logs"
+    cd build/*
+
+    local adoptLogArchiveDir="AdoptOpenJDKLogsArchive"
+
+    # Create new folder for failure logs
+    rm -rf ${adoptLogArchiveDir}
+    mkdir ${adoptLogArchiveDir}
+
+    # Copy build and failure logs
+    if [[ -f "build.log" ]]; then
+      echo "Copying build.log to ${adoptLogArchiveDir}"
+      cp build.log ${adoptLogArchiveDir}
+    fi
+    if [[ -d "make-support/failure-logs" ]]; then
+      echo "Copying make-support/failure-logs to ${adoptLogArchiveDir}"
+      mkdir -p "${adoptLogArchiveDir}/make-support"
+      cp -r "make-support/failure-logs" "${adoptLogArchiveDir}/make-support"
+    fi
+
+    # Find any cores, dumps, ..
+    find . -name 'core.*' -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' | sed 's#^./##' | while read -r dump ; do
```

Comment:
I expect the list of filename patterns here is incomplete. For the issue this was intended to address, it will be helpful to see files such as the following for `jdk.jshell`:
```
jdk/modules/jdk.jshell/_the.jdk.jshell_batch
jdk/modules/jdk.jshell/_the.jdk.jshell_batch.cmdline
jdk/modules/jdk.jshell/_the.jdk.jshell_batch.filelist
jdk/modules/jdk.jshell/_the.jdk.jshell_batch.log
jdk/modules/jdk.jshell/_the.jdk.jshell.config_vardeps
jdk/modules/jdk.jshell/_the.jdk.jshell_pubapi
jdk/modules/jdk.jshell/_the.jdk.jshell-server.conf
jdk/modules/jdk.jshell/_the.jdk.jshell.vardeps
```

Reply:
It does capture the whole of make-support/failure-logs, which will contain the above if it fails, I confirmed that with a test I did

Reply:
the above "dump" list is generic for hotspot/openj9/corretto...

Reply:
See test example here: https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-linux-x64-openj9-linuxXL/835/

Reply:
Ok, that will capture the log, but not `.cmdline` or `.filelist`.
Also, I don't see any clues in the log where one would find the resulting archive.

Reply:
The .cmdline is captured as that is placed in failure-logs:  
[OpenJDK11U-makefailurelogs_x64_linux_openj9_linuxXL_2021-02-12-15-16(1).tar.gz](https://github.com/AdoptOpenJDK/openjdk-build/files/5972992/OpenJDK11U-makefailurelogs_x64_linux_openj9_linuxXL_2021-02-12-15-16.1.tar.gz)

Reply:
The archive is archived same as where images etc would be:
![image](https://user-images.githubusercontent.com/31470007/107792728-5f143000-6d4d-11eb-947f-c38b66a2773e.png)

Reply:
Your earlier comment referenced build 835, which doesn't have a corresponding artifact that I could find. Did something change in between that build and 838?

Also, the `.cmdline` file available for 838 makes reference to
```
@/home/adoptopenjdk/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-x64-openj9-linuxXL/workspace/build/src/build/linux-x86_64-normal-server-release/jdk/modules/jdk.jshell/_the.jdk.jshell_batch.tmp
```
but `_the.jdk.jshell_batch.tmp` is not included in the archive.

Reply:
Yes, sorry Adopt only keeps 1 artifact history on the build job and a subsequent one was submitted, we're thinking of increasing this

Reply:
we'd need an upstream change if something is not being copied to the make-support/failed-logs folder

Reply:
> we'd need an upstream change if something is not being copied to the make-support/failed-logs folder

Not if you capture file matching `jdk/modules/*/_the.*`.

Reply:
ah see what you mean, thanks Keith